addons/website/static/src/js/content/snippets.animation.js

```diff
@@ -1148,6 +1149,349 @@ registry.anchorSlide = publicWidget.Widget.extend({
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Draws the whole countdown, including one countdown for each time unit.
+     *
+     * @private
+     */
+    _render: function () {
+        this._updateTimeDiff();
+        _.each(this.diff, (val, index) => {
+            var canvas = val['canvas'];
+            var ctx = canvas.getContext("2d");
+            ctx.canvas.width = this.size;
+            ctx.canvas.height = this.size;
+            this._clearCanvas(ctx);
+
+            var hideCanvas = !val['visible'] || (this.isFinished && !this.editableMode && this.$el.hasClass('hide-countdown'));
+            $(canvas).toggleClass('d-none', hideCanvas);
+            if (hideCanvas) {
+                return;
+            }
+
+            if (this.shapeBackground !== 'none') {
+                this._drawBgCircle(ctx, this.shapeBackground === 'plain');
+            }
+            this._drawText(canvas, val['nb'], val['label']);
+            if (this.progressBarStyle === 'surrounded') {
+                this._drawOuterCircleBg(ctx, this.progressBarWeight === 'thin');
+            }
+            if (this.progressBarStyle !== 'none') {
+                this._drawOuterCicle(ctx, val['nb'], val['total'], this.progressBarWeight === 'thin');
+            }
+            $(canvas).toggleClass('mx-2', this.shape === 'boxes');
+        });
+        if (this.isFinished) {
+            clearInterval(this.setInterval);
+            if (!this.editableMode) {
+                this._handleEndCountdown();
+            }
+        }
+    },
+    /**
+     * Updates the remaining units into the `diff` object.
+     *
+     * @private
+     */
+    _updateTimeDiff: function () {
+        var currentTimestamp = Date.now() / 1000;
+        var delta = this.endTime - currentTimestamp;
+
+        this.isFinished = delta < 0;
+        if (this.isFinished) {
+            this.diff['days']['nb'] = 0;
+            this.diff['hours']['nb'] = 0;
+            this.diff['minutes']['nb'] = 0;
+            this.diff['seconds']['nb'] = 0;
+            return;
+        }
+
+        this.hereBeforeTimerEnds = true;
+
+        if (this._isUnitVisible('d')) {
+            var days = Math.floor(delta / 86400);
+            delta -= days * 86400;
+            this.diff['days']['nb'] = days;
+        }
+        if (this._isUnitVisible('h')) {
+            var hours = Math.floor(delta / 3600) % 24;
+            delta -= hours * 3600;
+            this.diff['hours']['nb'] = hours;
+        }
+        if (this._isUnitVisible('m')) {
+            var minutes = Math.floor(delta / 60) % 60;
+            delta -= minutes * 60;
+            this.diff['minutes']['nb'] = minutes;
+        }
+        if (this._isUnitVisible('s')) {
+            var seconds = parseInt(delta);
+            this.diff['seconds']['nb'] = seconds;
+        }
```

Comment:
Here, on each rendering you recheck if each unit is visible (by looping over a string)... but this never change. The code can be generic with the array I mentioned earlier:
```javascript
for (const unitData in this.difff) {
      unitData.nb = Math.floor(delta / unitData.nbSeconds);
      delta -= unitData.nb * unitData.nbSeconds;
}
```

Reply:
Done, better that way indeed.